lib/rouge/lexers/ecl.rb

```diff
@@ -0,0 +1,136 @@
+        of only opt outer overwrite packed partition penalty 
+        physicallength pipe quote record repeat return right rows 
+        scan self separator service shared skew skip sql 
+        store terminator thor threshold token transform trim type 
+        unicodeorder unsorted validate virtual whole wild within xml 
+        xpath after cluster compressed compression default encoding escape 
+        fileposition forward grouped inner internal linkcounted literal lzw 
+        mofn multiple namespace wnotrim noxpath onfail prefetch retry 
+        rowset scope smart soapaction stable timelimit timeout unordered 
+        unstable update use width
+      )
+
+      functions = %w(
+        abs acos aggregate allnodes apply ascii asin asstring 
+        atan _token ave case catch choose choosen choosesets 
+        clustersize combine correlation cos cosh count covariance cron 
+        dataset dedup define denormalize dictionary distribute distributed distribution 
+        ebcdic enth error evaluate event eventextra eventname exists 
+        exp failcode failmessage fetch fromunicode fromxml getenv getisvalid 
+        global graph group hash hashcrc having httpcall httpheader 
+        if iff index intformat isvalid iterate join keyunicode 
+        length library limit ln local log loop map 
+        matched matchlength matchposition matchtext matchunicode max merge mergejoin 
+        min nofold nolocal nonempty normalize parse pipe power 
+        preload process project pull random range rank ranked 
+        realformat recordof regexfind regexreplace regroup rejected rollup round 
+        roundup row rowdiff sample set sin sinh sizeof 
+        soapcall sort sorted sqrt stepped stored sum table 
+        tan tanh thisnode topn tounicode toxml transfer transform 
+        trim truncate typeof ungroup unicodeorder variance which workunit 
+        xmldecode xmlencode xmltext xmlunicode apply assert build buildindex 
+        evaluate fail keydiff keypatch loadxml nothor notify output 
+        parallel sequential soapcall wait
+      )
+
+      class1 = %w(
+        file date str math metaphone metaphone3 uni audit blas system
+      )
+
+      class2 = %w(
+        debug email job log thorlib util workunit
+      )
+
+      state :single_quote do
+        rule %r([xDQUV]?'([^'\\]*(?:\\.[^'\\]*)*)'), Str::Single
+        rule %r/\\(x\\h{2}|[0-2][0-7]{,2}|3[0-6][0-7]?|37[0-7]?|[4-7][0-7]?|.|$)/, Text
+      end
+
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/[*].*?[*]/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text
+        rule %r(//.*), Comment::Single
+        mixin :inline_whitespace
+      end
+
+      state :root do
+	mixin :whitespace
+	mixin :single_quote
+
+        rule %r(\b(?i:(begin|end)c\+\+)), Keyword
+        rule %r(\b(?i:(#{keywords.join('|')}))\b), Keyword
```

Comment:
Rather than have a large number of similar rules that match against various words, we generally encourage a single rule where the match is checked using Ruby's Array/Set inclusion predicates. You can see how this can work in the JavaScript lexer as well:

https://github.com/rouge-ruby/rouge/blob/39b0ed7117a421fea1cd1403e8a4e17844653bc7/lib/rouge/lexers/javascript.rb#L176-L192

Reply:
Yes. I was aware of that. I was having problem with including words like "beginc++" with special characters. Now that I have separated them, I can do that.